### ISSUE TITLE
Extract latest currency json according to one's local currency as base currency

### ISSUE THREAD

**Issue by Daytron** [Daytron/daytron-money#1]
Currently the default base currency is USD and a workaround calculation is done to convert non-USD currency to another.

The API recently released new set of features that will allow to change the default base currency when requesting a latest.json file. 

Allows `CurrencyExchange` to set a base currency and automatically update the currency rates HashMap.
- Decide on number of API requests allowed when changing base currency (currently only after 12 hours )
- keep `connectAndTryToUpdateCurrencyRates()` method on 12 hours limit
- Save previous base currency HashMap variables in order to limit dependency on api
  - `connectAndTryToUpdateCurrencyRates()` method can be used to update all base currency data
- `convert` method will follow strict direct conversion using given base currency rates for accuracy instead  of calculating the rate when converting non-base currency to another.
- Create `convertEstimate` method to convert non-base currency to another. This is maybe imprecise compare to actual rate using its base currency.

**Comment by Daytron** [comment]
The api is no longer available as of this writing. This feature is on hold for now and won't be part of 1.2 release.
